src/core/bittorrent/torrenthandle.cpp

```diff
@@ -1203,7 +1203,11 @@ void TorrentHandle::setFirstLastPiecePriority(bool b)
 
             QPair<int, int> extremities = fileExtremityPieces(index);
             SAFE_CALL(piece_priority, extremities.first, prio);
-            SAFE_CALL(piece_priority, extremities.second, prio);
+            // worst case: AVI index = 1% of total file size
+            int nLastPieces = ceil(fileSize(index) * 0.01 / pieceLength());
+            for (int i = 0; i < nLastPieces; ++i)
+                SAFE_CALL(piece_priority, extremities.second - i, prio);
```

Comment:
It seems to me that @ngosang very carefully didn't change the priority call itself, instead simply adding a calculated variable for which pieces get affected by that call.  Otherwise I suspect he would have had to rewite this entire section. 

Perhaps you could show an example of how to implement your suggestion into what's already here?